### PR TITLE
Adjust KHR_texture_transform offset for non-zero rotation center

### PR DESCRIPTION
**Pull request by alexchuber**
Forum issue: https://forum.babylonjs.com/t/sandbox-the-khr-texture-transform-property-of-gltfexport-cannot-be-exported/55425

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/16015/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16015/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16015/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/16015/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/16015/merge
https://gui.babylonjs.com/?snapshot=refs/pull/16015/merge
https://nme.babylonjs.com/?snapshot=refs/pull/16015/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/16015/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16015/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16015/merge/testResults/webgpuplaywright/index.html

File: packages/dev/serializers/src/glTF/2.0/Extensions/KHR_texture_transform.ts [hunk 1]
```diff
@@ -56,6 +66,10 @@ export class KHR_texture_transform implements IGLTFExporterExtensionV2 {
             if (babylonTexture.wAng !== 0) {
                 textureTransform.rotation = -babylonTexture.wAng;
                 transformIsRequired = true;
+
+                if (babylonTexture.uRotationCenter !== 0 || babylonTexture.vRotationCenter !== 0) {
+                    textureTransform.offset = AdjustOffsetForRotationCenter(babylonTexture);
+                }
```
Comment (bghgary) [inline packages/dev/serializers/src/glTF/2.0/Extensions/KHR_texture_transform.ts:72]: I think this will work if we can guarantee these properties are not being animated by KHR_animation_pointer, but it's not easy to determine if they are.

Comment (bghgary) [reply packages/dev/serializers/src/glTF/2.0/Extensions/KHR_texture_transform.ts:72]: For now, maybe log a warning when this happens?

Comment (bghgary) [reply packages/dev/serializers/src/glTF/2.0/Extensions/KHR_texture_transform.ts:72]: We also have to be a bit careful with the math. The spec itself has some issues and we have to make sure the order of operations matches correctly. See https://github.com/KhronosGroup/glTF/issues/1563.

Comment (alexchuber) [reply packages/dev/serializers/src/glTF/2.0/Extensions/KHR_texture_transform.ts:72]: We don't yet support KHR_animation_pointer export, but I do see how this could be problematic once we do. 
A warning sounds good for now.

As for the math piece, I tried to take care with the order of operations & preserved the negative rotation we use to comply with glTF spec. I did some testing, but let me do a bit more to be sure.

Comment (alexchuber) [reply packages/dev/serializers/src/glTF/2.0/Extensions/KHR_texture_transform.ts:72]: Tested more for peace of mind, math should be good. I'd add a visual test, but it might be too much for such an edge case like this?

**Review by bghgary — APPROVED** [review]
LGTM

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16015/merge/testResults/webgpuplaywright/index.html
